### PR TITLE
Make ManiSkill2 pip installable

### PR DESCRIPTION
**Pull request by Jiayuan-Gu**
- `mani_skill2.assets` contain necessary robot descriptions and task assets, which will be packaged into the wheel.
- Use `mani_skill2.PACKAGE_ASSET_DIR` to manage assets that will be packaged into wheels, and use `mani_skill2.ASSET_DIR` to manage assets require downloading.
- Refactor `download` to `download_asset`, and support downloading by group and environment id. For example, `python -m mani_skill2.utils.download_asset PickClutterYCB-v0`

### CODE REVIEW

File: mani_skill2/utils/download_asset.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        "uid",
+        type=str,
+        help="Asset UID. Use 'all' to download all assets.",
+    )
+    parser.add_argument("--quiet", action="store_true", help="Disable verbose output.")
+    parser.add_argument(
+        "--non-interactive", action="store_true", help="Disable prompts."
+    )
+    return parser.parse_args()
+
```
Comment (StoneT2000) [inline mani_skill2/utils/download_asset.py:221]: Should we add something to skip re-downloading already downloaded assets? might make running things a bit easier without having to check yourself with workloads that don't spin new instances all the time.

might also want to do this with datasets altho versioning is a problem perhaps

File: setup.py [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
+        exclude=["warp_maniskill.warp.tests"],
+    ),
+    python_requires=">=3.7",
+    install_requires=read_requirements(),
+    # Glob patterns do not automatically match dotfiles
+    package_data={
+        "mani_skill2": ["assets/**", "envs/mpm/shader/**"],
+        "warp_maniskill.warp": ["native/*", "native/nanovdb/*"],
+    },
+    exclude_package_data={"": ["*.convex.stl"]},
```
Comment (StoneT2000) [inline setup.py:32]: have you tried to package this to the test pypi server yet? is there smth I can test with / build with?

**Review by StoneT2000 — CHANGES_REQUESTED** [review]

File: mani_skill2/utils/download_asset.py (hunk 1 above)
Comment (Jiayuan-Gu) [reply mani_skill2/utils/download_asset.py:221]: Skip re-downloading is already implemented (but it requires users to decide).

The old assets are still kept for backward compatibility. The versioning can somehow be implemented like Habitat, but may need some re-designing.

File: setup.py (hunk 2 above)
Comment (Jiayuan-Gu) [reply setup.py:32]: If you want to test it locally, you can just `pip install .`

**Pull request merged**